### PR TITLE
[BottomNavigation] Add `barTintColor` to replace `backgroundColor`

### LINKED ISSUE #2832: [BottomNavigation] backgroundColor UIAppearance is disabled
### Overview
The `init` method for BottomNavigation disabled UIAppearance on its `backgroundColor` property by assigning a value to the setter. This prevents clients from theming it via UIAppearance.

### Reproduction steps
1. Create a `setBackgroundColor:` method in MDCBottomNavigationBar.m with a breakpoint
1. Set the `backgroundColor` property via UIAppearance
1. Watch the breakpoint

### Version number
44.3.0

### Operating system and device
Any/All

### PR DESCRIPTION
**Pull request by romoore**
Here's how the default catalog theme will look:

mdc-bottomnav-catalog-theme

Closes #2832

### CODE REVIEW

File: components/BottomNavigation/src/MDCBottomNavigationBar.m [hunk 1]
```diff
@@ -513,6 +523,20 @@ - (void)setItemTitleFont:(UIFont *)itemTitleFont {
...
 }
 
+- (void)setBarTintColor:(UIColor *)barTintColor {
+  _barTintColor = barTintColor;
+  self.backgroundColor = barTintColor;
+}
+
+- (void)setBackgroundColor:(UIColor *)backgroundColor {
+  super.backgroundColor = backgroundColor;
+  _barTintColor = backgroundColor;
```
Comment (ianegordon) [inline components/BottomNavigation/src/MDCBottomNavigationBar.m]: Does this behavior match UITabBar / UINavigationBar?

**Review by ianegordon — DISMISSED** [review]

File: components/BottomNavigation/src/MDCBottomNavigationBar.m (hunk 1 above)
Comment (romoore) [reply components/BottomNavigation/src/MDCBottomNavigationBar.m]: For both UITabBar and UINavigationBar, setting the `barTintColor` property changes the color of the background of the view.  

## UIKit

**UITabBar.h**
```objectivec
/*
 The behavior of tintColor for bars has changed on iOS 7.0. It no longer affects the bar's background
 and behaves as described for the tintColor property added to UIView.
 To tint the bar's background, please use -barTintColor.
 */
@property(nullable, nonatomic, strong) UIColor *barTintColor NS_AVAILABLE_IOS(7_0) UI_APPEARANCE_SELECTOR;  // default is nil
```

**UINavigationBar.h**
```objectivec
/*
 The behavior of tintColor for bars has changed on iOS 7.0. It no longer affects the bar's background
 and behaves as described for the tintColor property added to UIView.
 To tint the bar's background, please use -barTintColor.
 */
@property(nullable, nonatomic,strong) UIColor *barTintColor NS_AVAILABLE_IOS(7_0) UI_APPEARANCE_SELECTOR;  // default is nil
```

## MDC-iOS

**MDCTabBar.h**

```objectivec
/**
 Tint color to apply to the tab bar background.

 If nil, the receiver uses the default background appearance. Default: nil.
 */
@property(nonatomic, nullable) UIColor *barTintColor UI_APPEARANCE_SELECTOR;
```

File: components/BottomNavigation/src/MDCBottomNavigationBar.h [hunk 2]
```diff
@@ -108,6 +108,16 @@ typedef NS_ENUM(NSInteger, MDCBottomNavigationBarAlignment) {
...
 
+/**
+ Color of the background of the MDCBottomNavigationBar.
+ */
+@property(nonatomic, strong, nullable) UIColor *barTintColor UI_APPEARANCE_SELECTOR;
+
+/**
+ To color the background of the view use -barTintColor instead.
+ */
+@property(nullable, nonatomic, copy) UIColor *backgroundColor UI_APPEARANCE_SELECTOR;
```
Comment (ianegordon) [inline components/BottomNavigation/src/MDCBottomNavigationBar.h]: copy?  Seriously Apple?.....sigh.

**Review by ianegordon — APPROVED** [review]

**Comment by ianegordon** [comment]
Please resolve conflicts before merge.

**Pull request merged**